src/services/services.ts

```diff
@@ -1086,6 +1088,12 @@ namespace ts {
         newText: string;
     }
 
+    export interface TextInsertion {
+        newText: string;
+        /** The position in newText the caret should point to after the insertion. */
+        caretOffset: number;

```

Comment:
Actually it would potentially be interesting if this was a series of ranges that one could tab through.